src/components/booksform.js

```diff
@@ -1,32 +1,87 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import Proptypes from 'prop-types';
+import { createAction } from '../actions/index';
+import randomNumber from '../logic';
 
-const BooksForm = () => {
-  const bookCategories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
-  return (
-    <form>
-      <fieldset className="form-group">
-        <label htmlFor="formInput">
-          title
-          <input
-            type="text"
-            className="form-control"
-            placeholder="Enter a book Title"
-            id="formInput"
-          />
-        </label>
-      </fieldset>
-      <fieldset>
-        <select>
-          {bookCategories.map(category => (
-            <option key={category} value={category}>
-              {category}
-            </option>
-          ))}
-        </select>
-      </fieldset>
-      <button type="submit">Submit</button>
-    </form>
-  );
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    if (e.target.id === 'formInput') {
+      this.setState({ title: e.target.value });
+    } else {
+      this.setState({ category: e.target.value });
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { addBook } = this.props;
+    const inputField = document.getElementById('formInput');
+
+    const book = {
+      id: randomNumber(),
+      ...this.state,
+    };
+    addBook(book);
+    this.setState({
+      title: '',
+      category: '',
+    });
```

Comment:
normally `category` should be set back to the DEFAULT category instead of empty string

Reply:
This issue has now been rectified.